First Step in adding carousel/caption functionality

rowcla opened the pull request.

Chewyfitz committed:
Filling out function of Album GET, splitting off functionality of routes into /util/ files.

Chewyfitz committed:
server.js - added comments for possible expansion routes;
routes/album.js - removed unnecessary database.js reference;
routes/image.js - updated POST route, removed database.js reference;
util/image.js - added `addPhotoToUser` function;
added user.js route and util.

Chewyfitz committed:
Merge branch 'master' of https://github.com/Chewyfitz/comp30022-it-project into Back-End

# Conflicts:
#	back-end/routes/album.js
#	back-end/routes/image.js
#	back-end/server.js
#	client/src/index.js

Chewyfitz committed:
Merge branch 'master' into Back-End

PCA000 committed:
Just a 'quick' update

Added functionality to retieve a range of photos from the database (currently slow)
Working on a faster prototype but am strugling a bit with it if I'm honest.
Changed the return value of a fail from null to undefined (The code still passes my simple tests so it shouldn't have broken too badly if at all)

Chewyfitz committed:
Back-end GCS upload (#7)

* Adding Google Cloud Storage photo uploads (thanks to Rowan!)

* Updated a little bit, still yet to test.

* Fiddling with return values - Beware: GCS initialisation creates a file that MUST NOT be added to the git

* cleanup, reorganised the file upload so it happens concurrently instead of sequentially.

* Update index.js

* Create UploadButton.js

* Update index.js

* Update index.js (+1 squashed commits)

Squashed commits:

[439e43b] Create UploadButton.js (+1 squashed commits)

Squashed commits:

[67f3397] Update index.js

* Reverse mistaken commit

* Reverse Mistaken Commit (again because git weirdness)

Chewyfitz committed:
Deleted some files before merging so they don't get added to the master branch

Chewyfitz committed:
Replacing deleted file for smooth merging

PCA000 committed:
Some documentation has been updated:
    - int type corrected to Number

Added functionality to get all the photo data from a User:
    - getAllPhotoData(userID)

Added two ways to retrieve a range of albumPosition docs:
    - getSomeAlbumPhotos(userID, albumID, start, end)
    - getAllAlbumPhotos(userID, albumID, start, end)

Did some optimisation on the getAllAlbumPhotos:
    - For an album size of 200 retrieving 10 the results are resonably close (getSomeAlbumPhotos is still quicker)
    - Relative speeds untested for single document retrieval (should be using single document functions for that anyway)

Notes: I may have forgoten to export some functions as per usual. Testing is yet to be done.

PCA000 committed:
Merge branch 'Back-End' of https://github.com/Chewyfitz/comp30022-it-project into Back-End

PCA000 committed:
Implemented single Document deletion:
    - deleteDoc(path, doc)
Implemented single layer Collection deletion:
    - deleteCollection(path)

Prototyped a Complete Document Deletion
    - NOT YET IMPLEMENTED
Prototyped a Complete Collection Deletion
    - NOT YET IMPLEMENTED

Chewyfitz committed:
Clearing out Back-end client to merge master into this branch

Chewyfitz committed:
Merge branch 'master' into Back-End

# Conflicts:
#	.gitignore
#	back-end/database.js
#	back-end/database/album_pages.js
#	back-end/database/albums.js
#	client/package-lock.json
#	client/public/index.html
#	client/src/App.css
#	client/src/app.js
#	client/src/components/layout/navbar/NavDropdown.js
#	client/src/components/layout/navbar/Navbar.css
#	client/src/components/layout/navbar/Navbar.js
#	client/src/components/layout/navbar/Sidebar.js
#	client/src/components/layout/navbar/SubNavbar.js
#	client/src/components/layout/photolist/PhotoList.css
#	client/src/components/layout/photolist/PhotoList.js
#	client/src/index.js

Chewyfitz committed:
Implement Testing into Back-end branch (#12)

* Added a testing file, and a sample (test.js) to be able to create more testing files. Currently the /test/api_image_test.js does not actually invoke any tests, but the boilerplate is there and can be expanded upon.

* Added tests for some image routes, placeholders for the rest.
Added an export to server.js so that it can be loaded into the test script

* Added a file to test album routes.
Got some annoying console logging that makes the test prints a bit gross, but that's outside the scope of this branch.

* added done() clause to a test so it doesn't hang

* Updated test descriptions to explain the HTTP response codes instead of just "success".

PCA000 committed:
Implementation of deletion functionality

Added the functionality to delete documents. This functionality has been tested and exported.

Added:
deleteCollection
deleteDoc
recDeleteAllFromCollection
recDeleteAllFromDoc
deleteUser
deletePhoto
deleteAlbum
deleteAlbumPage
deleteAlbumPosition

Exported:
deleteUser
deletePhoto
deleteAlbum
deleteAlbumPage
deleteAlbumPosition

Usage: Has been kept consistent with the arguments and return values of other functions. The arguments take the needed document IDs and returns true only if the document and all of its subdocuments were successfully deleted from the database.

Also reorded the order of some functions to make things a bit easier to find.

Chewyfitz committed:
Adding authentication - setting up routes for dealing with users, not enforcing authentication yet

Chewyfitz committed:
Added login route - uses the authorization header

Chewyfitz committed:
Back end testing create update (#14)

* Added album Post, Put, and Patch test cases

* Added album page Get test

* Added image URL Post test

* Added api user test file with structure to write tests

Chewyfitz committed:
/api/login and /api/register (alias /api/signup) both should be in working order. Let me know if you run into problems

Chewyfitz committed:
implemented GETing a user. This very much should not be expected to work as it's a little bit out of scope

Chewyfitz committed:
Added functionality for sending password reset emails. Still missing user details update, user delete, update email, send verification email, and change password.

PCA000 committed:
Config changes

Chewyfitz committed:
comments

Chewyfitz committed:
Back end route processing (#18)

* Added Create and Update requests to /api/album

* Added route processing for user {Create, Read}, and image/photo {Create, Read, Update}

* Adding framework for albumPages, added default case for returning a list of all albums

* Added albumPage route handling. Might error on some unexpected values, and has hard-coded user

* Updating the hard-coded user value to be a query variable instead

* Updating lots of typos, implemented getAllAlbums via empty album ID GET request to /album/

* Actually returning stuff from database function calls and properly responding on album page updates

Chewyfitz committed:
Merge branch 'Back-End' into backend/authentication

Chewyfitz committed:
Removing some unnecessary prints that probably shouldn't have been included

PCA000 committed:
Merge pull request #15 from Chewyfitz/database-deletion

Database deletion functionality

Chewyfitz committed:
Merge pull request #19 from Chewyfitz/backend/authentication

Backend/authentication

Chewyfitz committed:
hotfix: Adding Cors to the backend main branch

Chewyfitz committed:
Merge pull request #20 from Chewyfitz/backend/hotfix

hotfix: Adding Cors to the backend main branch

Chewyfitz committed:
An actual real hotfix - Breaking convention to push this change because it's a typo and breaks the whole app

rowcla committed:
Temporary print for debugging

Chewyfitz committed:
Revert "Temporary print for debugging"

This reverts commit 8c8edfa306ccce450ddb508800f2d89276e95bcb.

Chewyfitz committed:
Added the foundations for pagination of albums,
Modified album route so it actually returns multiple photos instead of just one
Modified image route so added photos are successfully added to the un-album

rowcla committed:
Another temporary testing update

Alternative test processes are likely too time consuming

rowcla committed:
Adding in options route for account creation

Time constraints require me to directly push to here. This shouldn't be an issue anyway

Chewyfitz committed:
Added PUT route for album to add an image to an album.

Chewyfitz committed:
add album DELETE route

Chewyfitz committed:
add albumPosition DELETE route handling

Chewyfitz committed:
add ablumPage DELETE route

Chewyfitz committed:
add image DELETE route handling

Chewyfitz committed:
Updated some comments etc/ refactoring code

Chewyfitz committed:
fixed a mistake in database/photos.js

Chewyfitz committed:
Changing functionality to use a database function that Patrick has already written

Chewyfitz committed:
fixing mistakes as pointed out by PCA000

Chewyfitz committed:
Merge pull request #24 from Chewyfitz/backend/delete

Backend/delete (untested)

Chewyfitz committed:
Merge branch 'Back-End' into backend/fix_album_GETs

Chewyfitz committed:
Fixing a couple of mistakes that were picked up by Rowan

Chewyfitz committed:
Adding a bunch of checks to ensure the user id is sent in the album router

Chewyfitz committed:
Adding pagination functionality to album GETs

Chewyfitz committed:
changing the check for success for adding an image to album due to the return value potentially being 0 (ie. false)

Chewyfitz committed:
Merge pull request #23 from Chewyfitz/backend/fix_album_GETs

Backend/fix album gets (poorly named branch)

Chewyfitz committed:
HOTFIX: Dealing with the function getSomeAlbumPhotos() having slightly buggy behaviour - replaced with a modified getAllAlbumPhotos as an interim solution

Chewyfitz committed:
HOTFIX: I'm dumb and it actually wasn't Patrick's function -- I'm just really bad at promises

Chewyfitz committed:
adding authentication method to send the password reset email

Chewyfitz committed:
creating a bodge way of creating the un-album until Patrick implements a proper one

Chewyfitz committed:
Merge pull request #30 from Chewyfitz/backend/user-gets-unalbum-hotfix

creating a bodge way of creating the un-album until Patrick implement…

PCA000 committed:
Override Album ID (#28)

PCA000 committed:
UnAlbum doc Key override (#32)

Adds the needed argument to the unalbum creation call in order override how album document keys are generated. The album document for the unalbum will now have the key 'un'
*Overwrites the temporary hack used in #31 with intended functionality*

Chewyfitz committed:
Merge pull request #27 from Chewyfitz/backend/fix-pw-reset

adding authentication method to send the password reset email

Chewyfitz committed:
Adding the one line to fix image ids not being returned

Chewyfitz committed:
Merge pull request #40 from Chewyfitz/backend/fix-albumPos-sends

Adding the one line to fix image ids not being returned

PCA000 committed:
Pull backend/database/Database-Master into  Back-End (#35)

Added:
Pixel Height and Pixel Width to the Photos Docs
getPhotoDimensions(userID, photoID) -> {height: Number, width: Number}
updatePhotoDimensions(userID, photoID, height, width) -> Boolean
addManyAlbumPosition(userID, albuID, photoInfoList) -> Boolean
updateAlbumPositionOrder(userID, albumID, permutation) -> Boolean

Updated:
Removed database/album.js debugging code to resolve #13 
general_database.js has now has proper overwrite in place, and logs any promise rejections and error.
addPhoto(userID, photoReference, photoDateTime=null) became addPhoto(userID, photoReference, height=1, width=1, photoDateTime=null)
addAlbum(userID, name, override=undefined)
return values of getSomeAlbumPhotos and getAllAlbumPhotos {position: {reference: String, caption: String}} became {position: {reference: String, caption: String, photoID: String}}
deleteAlbumPosition now implemented
JSDOC

Removed:
A lot of bugs! (please don't find any more bugs in this because all the easy ones to fix are gone XD )

Chewyfitz committed:
hotfix: param instead of query

Chewyfitz committed:
hotfix: if(0) is false...

Chewyfitz committed:
Hotfix: consistency (userId should just be user)

Chewyfitz committed:
Creating a route to add multiple images to an album.

Chewyfitz committed:
Merge pull request #42 from Chewyfitz/backend/add_multiple_albumpos

Creating a route to add multiple images to an album.

Chewyfitz committed:
Should be fixed

Chewyfitz committed:
Merge pull request #44 from Chewyfitz/frontend/fixing_AlbumPhotoList

Should be fixed

Chewyfitz committed:
imageList did not have a key value so was assigned one

Chewyfitz committed:
For some reason user creation does not create a proper user document. Adding a placeholder "email address" to deal with that for now.

Chewyfitz committed:
Resolving some conflicts that were still present for some reason

Chewyfitz committed:
Adding email to user creation function

Chewyfitz committed:
Merge pull request #46 from Chewyfitz/backend/storing_email

Adding email to user creation function

Chewyfitz committed:
Adding an option to the album route to update the caption for a given album position (photo-album link). Pretty simplistically using existing functions, some extra processing might be required.

PCA000 committed:
Back end patrick routes&util (#43)

* updated routes/image.js post and util/image.js addPhotoToUser to accommodate the new height and width fields stored in the database.

* changed how we access the authentication functionalities of Firebase (so far as I can test, it works, but if there's any issues this would probably be the first thing to be reverted)

* Added logic to return the Album Template if there is no Page Template to overwrite it.

Chewyfitz committed:
adding an 'uploading' message to the upload button to give the user some feedback

Chewyfitz committed:
many files changed here -
PrivateRoute.js - changed the way private routes work so that route properties are passed along
app.js - updated the routes to use the new PrivateRoute component
ALBUMIFY.js - modified to utilise the callback function
AlbumPhotoList - refactored to be a component

Added new component:
AlbumView.js - Combines AlbumPhotoList (Overview) and Pagination (Preview) into a single component with a shared state. Will be elaborated on later today

rowcla committed:
Merge pull request #48 from Chewyfitz/backend/update_albumpos_caption

merge backend/update_albumpos_caption into back-end

Doesn't seem to be any issues, and I need this for my functionality

Chewyfitz committed:
Another big batch of changes, and a compromise that can probably be fixed pretty easily:

MainPage.js - Deleted the sidebar to be moved to the correct file
Sidebar.js - implemented feature
api.js - restructured AddImagesToAlbum and CreateNewAlbum functions as they were previously in a broken state. Current implementation will wait for each thing to be added before moving on to the next one as the back-end route for adding many images cannot currently handle sending an array of images. This should be pretty easy to fix, but the current implementation works for now.

Chewyfitz committed:
Implementing add multiple images to multiple albums.
api.js - added getAlbumList function, AddImagesToAlbum: added default album to remove from and remove value
NavDropdown.js - fixed formatting, implemented add list of images to multiple albums (quite slow - maybe some visual feedback?)
Sidebar.js - abstracted getAlbumList()
MainPage.js - fixed SubNavbar initialisation

Chewyfitz committed:
Implementing ALBUMIFY on front end. This should be my final change
api.js - added reorderImages function **BACKEND DOES NOT SUPPORT THIS**
ALBUMIFY.js - added a call to api reorderImages, now takes albumId as parameter.
AlbumPhotoList.js - passes albumId to ALBUMIFY button
Pagination.js - fixed to actually render album print preview

Chewyfitz committed:
oops typo

Anightingale committed:
Merge pull request #52 from Chewyfitz/frontend/fixing_album_redirects

Frontend/fixing a bunch of shit

Chewyfitz committed:
adding an implementation for adding many images to album (foundation was there but it wasn't complete)

Anightingale committed:
rearranged and deleted files for better abstraction

Anightingale committed:
Merge pull request #51 from Chewyfitz/frontend/adding-upload-status-message

adding upload status message

rowcla committed:
Merge branch 'FinalFunctionalityHopefully' into Front-End

Chewyfitz committed:
Merge pull request #53 from Chewyfitz/backend/implementing_multi_routes

implementing add many images to album on the back end

Chewyfitz committed:
Reverse a merge that wasn't from a pull request which broke the deployment.

Chewyfitz committed:
Merge pull request #38 from Chewyfitz/Back-End

Prepare for merge conflicts

Chewyfitz commented:
Looks like you might have done this backwards...

Chewyfitz reviewed (changes requested):
This merge request is for 
`Front-end` into `FinalFunctionalityHopefully`, instead of 
`FinalFunctionalityHopefully` into `Front-end`.

There's nothing stopping you from doing this merge, but instead the one that should have had a pull request opened was merged manually, which subsequently broke the deployment and has been reversed.
I apologise for reversing your commit, but it's important that process is still followed even now that the project is past-due.

If you want to make this pull request my recommendation is to merge `Front-end` into `FinalFunctionalityHopefully` and then re-make all of your changes by manually reversing the things that were actually additions.

Chewyfitz committed:
Merge branch 'master-dev' into Front-End

(I really hope this doesn't break anything)

# Conflicts:
#	.gitignore
#	back-end/database.js
#	back-end/database/album_pages.js
#	back-end/database/album_positions.js
#	back-end/database/albums.js
#	back-end/database/general_database.js
#	back-end/database/photos.js
#	back-end/database/users.js
#	back-end/routes/album.js
#	back-end/routes/image.js
#	client/.gitignore
#	client/package.json
#	client/src/index.css
#	client/src/index.js

Chewyfitz committed:
Fixed a couple of references that broke from the merge for things that are no longer used.

Chewyfitz committed:
Made a whole bunch of tiny changes to files to remove all of the warning messages.
No warning messages should print now.

Most notably any `<a>` tags that had empty hrefs (`href='#'` or the like) have been replaced with either `<p>` tags or `<button>` tags.
Also notably, all unused references have been removed.

Chewyfitz committed:
Merge branch 'Front-End' into frontend/neaten-code

# Conflicts:
#	client/src/components/layout/Carousel.js - deleted

Chewyfitz committed:
Merge pull request #59 from Chewyfitz/Frontend/neaten-code

rearranged and deleted files for better abstraction

Chewyfitz committed:
Merge branch 'Front-End' into frontend/fixing_errors

# Conflicts:
#	client/src/app.js
#	client/src/components/auth/LoginPageNavbar.css
#	client/src/components/layout/album/albumoverview/AlbumPhotoList.js
#	client/src/components/layout/album/albumpreview/Pagination.js
#	client/src/components/layout/navbar/navcomponents/NavDropdown.js
#	client/src/components/layout/navbar/navcomponents/UploadButton.js
#	client/src/pages/MainPage.js

Chewyfitz committed:
Merge pull request #60 from Chewyfitz/frontend/fixing_errors

Fixing all the warnings during compilation. No server-side warning should occur now.

Chewyfitz committed:
Some cheeky CSS modifications that need to be updated so the styling is correct
